src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -3548,14 +3552,75 @@ private static bool DirectoryInfoHasChildItems(DirectoryInfo directory)
                 }
                 else // Copy-Item local
                 {
+                    if (Context != null)
+                    {
+                        if (Context.ExecutionContext.SessionState.PSVariable.Get(SpecialVariables.ProgressPreferenceVarPath.ToString()).Value is ActionPreference progressPreference)
+                        {
+                            if (progressPreference == ActionPreference.Continue)
+                            {
+                                Task.Run(() =>
+                                {
+                                    GetTotalFiles(path, recurse);
+                                });
+                                _copyStopwatch.Start();
+                            }
+                        }
+                    }
+
                     CopyItemLocalOrToSession(path, destinationPath, recurse, Force, null);
+                    if (_totalFiles > 0)
+                    {
+                        _copyStopwatch.Stop();
+                        var progress = new ProgressRecord(COPY_FILE_ACTIVITY_ID, " ", " ");
+                        progress.RecordType = ProgressRecordType.Completed;
+                        WriteProgress(progress);
+                    }
                 }
             }
 
             _excludeMatcher.Clear();
             _excludeMatcher = null;
         }
 
+        private void GetTotalFiles(string path, bool recurse)
+        {
+            bool isContainer = IsItemContainer(path);
+
+            try
+            {
+                if (isContainer)
+                {
+                    var dir = new DirectoryInfo(path);
+                    var enumOptions = new EnumerationOptions();
+                    enumOptions.IgnoreInaccessible = true;
+                    enumOptions.AttributesToSkip = 0;
+                    enumOptions.RecurseSubdirectories = recurse;
+
+                    foreach (var file in dir.EnumerateFiles("*", enumOptions))
```

Comment:
```suggestion
                    var enumOptions = new EnumerationOptions()
                    {
                        IgnoreInaccessible = true,
                        AttributesToSkip = 0,
                        RecurseSubdirectories = recurse
                    };

                    foreach (var file in Directory.EnumerateFiles(path, "*", enumOptions))
```

Reply:
had to revert use of `Directory.EnumerateFiles()` as that just returns strings for the file names and we need the file sizes